File: java/leetcode_java/src/RunningSumOf1DArray1480.java
import java.util.ArrayList;
import java.util.List;
import java.util.Vector;

public class RunningSumOf1DArray1480 {
    Vector<Integer> numbers;

    // Constructor method
    public RunningSumOf1DArray1480(Vector<Integer> numbers) {
        // Initialize instance variables
        this.numbers = numbers;
    }

    public List<Integer> calculateRunningSum() {
        //System.out.print("1480. Running Sum of 1D Array");

        //int[] numbers = {3, 1, 2, 10, 1};
        int running_sum = 0;
        List<Integer> result_list = new ArrayList<>();

        // print the numbers array
        System.out.print("input: ");
        System.out.println(numbers);

        for (int num : numbers) {
            //System.out.printf("%d ", num);
            running_sum += num;
            result_list.add(running_sum);
        }

        return result_list;
     }
}

File: ruby/leetcode_ruby/running_sum_of_1D_array_1480.rb
class RunningSumOf1DArray1480
    def initialize(numbers)
        @numbers = numbers
    end

    def calculate_running_sum()
        running_sum = 0
        result = []

        @numbers.each do |number|
            running_sum += number
            # "running_sum: #{running_sum}"
            result.append(running_sum)
        end

        return result
    end
end
